src/otk/utils.py

```diff
@@ -0,0 +1,102 @@
+import json
+from typing import TypeVar, Generic
+
+
+def _deep_convert(data):
+    """Recursively convert nested dicts and lists into HiddenAttrDict and HiddenAttrList."""
+    if isinstance(data, (HiddenAttrDict, HiddenAttrList)):
+        return data
+    if isinstance(data, dict):
+        return HiddenAttrDict({key: _deep_convert(value) for key, value in data.items()})
+    if isinstance(data, list):
+        return HiddenAttrList([_deep_convert(item) for item in data])
+    return data
+
+
+class HiddenAttrList(list):
+    def __init__(self, *args, **kwargs):
```

Comment:
Why not have `__setitem__`/`__getitem__` and the slice dunders here to mirror how stuff is done in the dict variant?

Reply:
Not sure if I got your question right, the answer might be that there is no multiple inheritance so things have to be done twice as I wanted it to really behave like a real `dict` also type wise